commit cff19d1056d9613dbbd967497423b148b5ae455e (refs/changes/94/1694/2)
Author: Kody Kantor <Kody.Kantor@gmail.com>
Date:   2017-03-21T20:01:15-05:00 (2 years, 7 months ago)
    
    ZAPI-779 vmapi crashed when hitting undefined vm state in updateVm request

diff --git a/lib/endpoints/vms.js b/lib/endpoints/vms.js
index 63fe178..f03752f 100644
--- a/lib/endpoints/vms.js
+++ b/lib/endpoints/vms.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright (c) 2016, Joyent, Inc.
+ * Copyright (c) 2017, Joyent, Inc.
  */
 
 /*
@@ -392,6 +392,12 @@ function updateVm(req, res, next) {
     var sync = req.params.sync;
     var error;
 
+    if (!req.vm) {
+        error = [ errors.missingParamErr('vm') ];
+        return next(new errors.ValidationFailedError('Invalid Parameters',
+            error));
+    }
+
     if (!action) {
         error = [ errors.missingParamErr('action') ];
         return next(new errors.ValidationFailedError('Invalid Parameters',
diff --git a/test/vms.full.test.js b/test/vms.full.test.js
index f8cce7e..fc81451 100644
--- a/test/vms.full.test.js
+++ b/test/vms.full.test.js
@@ -838,6 +838,21 @@ exports.check_reboot_vm_nics_running = function (t) {
     });
 };
 
+// This test exists to prevent regression of ZAPI-779 where a user could
+// send a POST request without a VM UUID
+exports.undefined_vm_action = function (t) {
+    var params = {
+        action: 'add_nics'
+    };
+
+    var opts = createOpts(vmLocation, params);
+
+    client.post(opts, params, function (err, req, res, body) {
+        t.equal(res.statusCode, 409, '409 Conflict');
+        t.done();
+    });
+};
+
 
 exports.add_nics_with_networks = function (t) {
     var params = {
